packages/workspace/src/utils/version-utils.ts

```diff
@@ -0,0 +1,21 @@
+import { clean } from 'semver';
+
+export function checkAndCleanWithSemver(version: string) {
+  let newVersion = version;
+
+  if (clean(newVersion)) {
+    return newVersion;
+  }
+
+  if (version.startsWith('~') || version.startsWith('^')) {
+    newVersion = version.substring(1);
+  }
+
+  if (!clean(newVersion)) {
+    throw new Error(
+      `Unable to validate the provided version against semver - (${version})`
```

Comment:
This is a pretty unfriendly error message. Is it worth catching this in the migration and logging out something friendlier like "Your package.json lists a version of @angular/cli that Nx is unable to validate - (${version})"?